### DISCUSSION TITLE
How to validate a JWT signed by symmetric signer

### DISCUSSION THREAD

**Discussion by imdhemy** [lcobucci/jwt#903] [Q&A] [answered]
**Context**

Creating and validating a JWT token using Symmetric signer configuration.

**Error**
On validating the key with `SignedWith` constraint, the following error is thrown:

```
Lcobucci\JWT\Signer\InvalidKeyProvided : It was not possible to parse your key, reason:
* error:0909006C:PEM routines:get_name:no start line
* error:0909006C:PEM routines:get_name:no start line
```

**Source code**

```php
 $signer = new Sha256();
 $key = InMemory::file(__DIR__ . '/../../fixtures/private.key');
 $config = Configuration::forSymmetricSigner($signer, $key);

// The following line works as expected
 $token = $config->builder()->getToken($config->signer(), $config->signingKey());

 $validator = $config->validator();
 $signedWith = new SignedWith($signer, $config->signingKey());

// The exception is thrown here
 $this->assertTrue($validator->validate($token, $signedWith));
```

**Question**
How to validate such JWT?

**Comment by Ocramius** [comment]
Seems like your key is malformed? Please carefully read your exception message.

**Reply by imdhemy** [reply]
@Ocramius 

Correct me if I'm wrong.

I'm using the same key for Signing and validation (Symmetric). Generating the token works as expected, which means that the key is not malformed.

**Comment by SvenRtbg** [comment, marked as answer]
Seems like you think you are using Hmac Sha256 but you imported RSA Sha256.

**Reply by imdhemy** [reply]
@SvenRtbg Yes 🫢. Thanks a lot!
